runtime/libia2/threads.c

```diff
@@ -96,3 +94,100 @@ int __wrap_pthread_create(pthread_t *restrict thread,
   thread_thunk->data = data;
   return __real_pthread_create(thread, attr, ia2_thread_begin, thread_thunk);
 }
+
+// It's much simpler to only support a static number of created threads,
+// especially because we want to have very few dependencies.
+// If a program needs more threads, you can just increase this number.
+#define MAX_THREADS_SUPPORTED 4096
```

Comment:
We already limit the number of threads you can spin up. It's `IA2_MAX_THREADS` macro defined in ia2_internal.h

Reply:
Ahh okay, I'll use that then.

Reply:
@fw-immunant `IA2_MAX_THREADS` isn't explicitly referenced anywhere but we do limit the number of threads we can create right?

Reply:
Fixed in 09ee68344.

Reply:
`IA2_MAX_THREADS` is not actually a limit we ever needed. I think it was added as part of work towards a different design of per-compartment stacks that didn't use TLS, but TLS ultimately worked better and that never shipped. I would say we should just remove the `#define`, but if we're going to use it in permissive mode maybe we should rename it to clarify that it only affects permissive mode.

Reply:
Moved to #571.